### PR TITLE
Remove backend repos

### LINKED ISSUE #432: Split backends into independent repos
Discussion started here: https://github.com/stac-utils/stac-fastapi/issues/409#issuecomment-1192699464.

Including backends in this repo was convenient when stac-fastapi was first created but the pros are now outweighed by the cons.

Migration plan:
1. Merge all open PRs.
2. Release a new version, this will be the last version with a monorepo.
3. Create two new repos in `stac-utils` org, one for each backend.
4. Copy backend submodules over to new repos.
5. Merge the `api`, `extensions`, and `types` submodules into a single stac-fastapi package, keep import paths the same to avoid breaking.
6. Cut over to the new stac-fastapi package in the backend repos.
7. Test / validate backend repos before we merge all of the code.
8. Move backend specific issues to the appropriate repo.

### PR DESCRIPTION
**Pull request by gadomski**
**Related Issue(s):**

- Closes #432 
- Closes #450, #453, and #472 as OBE 
- Part of https://github.com/stac-utils/stac-fastapi/discussions/517
- Backend git history was extracted with https://github.com/gadomski/stac-fastapi-breakout/

**Description:**

No functional code changes. Removes the two backends, they live on their own now:

- https://github.com/stac-utils/stac-fastapi-pgstac
- https://github.com/stac-utils/stac-fastapi-sqlalchemy

It'd be nice to have a in-memory backend that we could use to exercise the API w/o a database, but that's a pretty significant lift. Something for the future, perhaps.

This is such a major change to the infrastructure (not the code) that I'd like to get at least two approving reviews on it.

**PR Checklist:**

- [x] `pre-commit` hooks pass locally
- [x] Tests pass (run `make test`)
- [x] Documentation has been updated to reflect changes, if applicable, and docs build successfully (run `make docs`)
- [x] Changes are added to the [CHANGELOG](https://github.com/stac-utils/stac-fastapi/blob/main/CHANGES.md).
- [ ] (once approved) Remove branch protections that reference the backends

### CODE REVIEW

**Review by moradology — APPROVED** [review]
Everything looks good to me and the downstream repos are working as expected. There's a minor typo I caught in the README, but otherwise LGTM

**Comment by gadomski** [comment]
> I'm noticing a few spots in which old references exist in the docs. 

Did you re-build the docs? I just did, and I'm not seeing any (all?) of those examples. The API docs are auto-generated.

I do see some dangling references in `tips-and-tricks`, will fix those.

**Comment by moradology** [comment]
I think you're right. I was misinterpreting the results of that grep

**Review by mmcfarland — APPROVED** [review]
These changes look good to me. I think there's one more file, `scripts/wait-for-it.sh` that is no longer used and could also be deleted.

As you mentioned in the PR, without a default backend, it's a little difficult to run/test changes without coordinating with a new backend repo. Do you have general (or specific) ideas about what the best dev workflow for updating this project with a backend? As a proof of concept, I tried mimicking a git submodule approach: cloning `stac-fastapi` into a gitignored subdirectory of `stac-fastapi-pgstac` and modifying the dockerfile there to do an editable install of these packages. Seemed a bit clumsy but should work. Perhaps when we've got a good flow we can update the development documentation here.

**Review by pjhartzell — COMMENTED** [review]
Just a question about the docs.

**Comment by gadomski** [comment]
> Perhaps when we've got a good flow we can update the development documentation here.

Agreed.

> Do you have general (or specific) ideas about what the best dev workflow for updating this project with a backend?

In the past, I've used my IDE's "workspace" concept, coupled with a virtual environment that has both packages installed in editable mode, to work on two repos at once. I think a similar approach could work here. That's relies more on a non-Docker development environment, so your submodule approach also seems good.

File: mkdocs.yml [hunk 1]
```diff
@@ -2,78 +2,65 @@ site_name: stac-fastapi
...
 # Social links
 extra:
   social:
-    - icon: 'fontawesome/brands/github'
-      link: 'https://github.com/stac-utils'
+    - icon: "fontawesome/brands/github"
+      link: "https://github.com/stac-utils"
 
 # Layout
 nav:
```
[the remark this answers is not in the document]
Comment (gadomski) [reply mkdocs.yml:16]: Good catch, thanks @pjhartzell 🙇🏼.

**Pull request merged**